### PR TITLE
nrf_wifi: Restructuring

### PR DESCRIPTION
**Pull request by krish2718**
1. Move `nrf_wifi` out of `hal_nordic` to it's own repo
2. Add a `modules/nrf_wifi` and move all the OSAL handling from Zephyr driver

### CODE REVIEW

**Review by sachinthegreen — DISMISSED** [review]

File: submanifests/optional.yaml [hunk 1]
```diff
@@ -27,6 +29,12 @@ manifest:
       remote: upstream
       groups:
         - optional
+    - name: nrf_wifi
```
Comment (carlescufi) [inline submanifests/optional.yaml:32]: @krish2718 as per a discussion yesterday in the TSC, please move this to the root's `west.yml`.

**Review by sachinthegreen — DISMISSED** [review]

**Review by jukkar — DISMISSED** [review]

**Review by jukkar — DISMISSED** [review]

**Review by rlubos — APPROVED** [review]

**Review by sachinthegreen — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Comment by carlescufi** [comment]
Removed DNM, since the added project has been approved by the TSC.

**Review by carlescufi — APPROVED** [review]

**Comment by jurenat** [comment]
What is the **we** file added in the root (https://github.com/zephyrproject-rtos/zephyr/pull/81101/commits/bb2dfa1c354ae9c7f8441bb1d1385cbbe310da95)? It looks like a mistake to me.

**Comment by krish2718** [comment]
> What is the **we** file added in the root ([bb2dfa1](https://github.com/zephyrproject-rtos/zephyr/commit/bb2dfa1c354ae9c7f8441bb1d1385cbbe310da95))? It looks like a mistake to me.

Oops, it is a mistake, will fix it, thanks.

**Comment by krish2718** [comment]
> > What is the **we** file added in the root ([bb2dfa1](https://github.com/zephyrproject-rtos/zephyr/commit/bb2dfa1c354ae9c7f8441bb1d1385cbbe310da95))? It looks like a mistake to me.
> 
> Oops, it is a mistake, will fix it, thanks.

FYI https://github.com/zephyrproject-rtos/zephyr/pull/82023

**Pull request merged**